include/swift/AST/Expr.h

```diff
@@ -4687,6 +4699,53 @@ class ApplyExpr : public Expr {
     Bits.ApplyExpr.NoAsync = noAsync;
   }
 
+  const DeclContext *getDeclContext() const {
+    assert(declContext
+           && "declContext must be initialized with setDeclContext");
+    return declContext;
+  }
+  void setDeclContext(const DeclContext *ctxt) {
+    declContext = ctxt;
+  }
+
+  bool makesUnsatisfiedIsolationJump() const {
```

Comment:
I wonder if we can come up with a better name here. `crossesIsolationDomain`? Perhaps even package this and the functions that follow it into a single function that returns an optional value that gives all of this information, e.g.,

```
llvm::Optional<IsolationCrossing> crossesIsolationDomain() const { ... }
```

where `IsolationCrossing` is some struct that has the caller and callee isolation domains. Maybe later we decide to move information about implicit async and such into it.